File: src/main/java/project/models/Reservation.java
package project.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import project.user.User;

import javax.persistence.*;
import java.io.Serializable;
import java.sql.Date;
import java.sql.Time;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "reservation")
public class Reservation implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private Date date;  // dia do mês ( ex: 20/3/ - dia 20 do mes 3 (Março))
    private Time time;  // horas da marcação ( 16:30:00 )


    @ManyToOne
    @JoinColumn(name="users_id", nullable=false)
    private User users;

    @ManyToOne
    @JoinColumn(name="service_id", nullable=false)
    private Service services;





}

File: mobile_app/lib/models/Reservation.dart
import 'package:mobile_app/models/Service.dart';
import 'package:mobile_app/models/User.dart';


class Reservation{
  String date;
  String time;
  int id;
  User user;
  Service service;



  Reservation();

  Reservation.post(String date, String time, User user, Service service){
    this.date=date;
    this.time=time;
    this.user=user;
    this.service=service;
  }

  Reservation.fromMap(Map<String, dynamic> data){
    id=data['id'];
    date=data['date'];
    time=data['time'];
    user=User.fromMap(data['users']);
    service=Service.fromMap(data['services']);
  
  }

}
